lib/elixir/src/elixir_map.erl

```diff
@@ -148,11 +148,13 @@ load_struct(Meta, Name, Args, InContext, E) ->
           form_error(Meta, ?key(E, file), ?MODULE, {undefined_struct, Name, Arity})
       end;
 
+    error:{badkey,Key} ->
+      form_error(Meta, ?key(E, file), ?MODULE, {unknown_key_for_struct, Name, Key});
```

Comment:
Perfect. But now I think we can get rid of this clause altogether. What do you think?

Reply:
I tested it without this clause, and then you get KeyError, but no line number info. Is it maybe better to change the catch-all clause at the end to also use `form_error`?

Reply:
Actually I'm not sure how that would work ... I'm not sure whether it's possible to write the general case to give a good error message when you try to use a key that doesn't exist.

Reply:
`KeyError (key :age not found)` is what we see if we remove this clause.

Reply:
I stand corrected, the stacktrace is good :joy:

Reply:
I'm actually of two minds on this. If we remove this clause then we get an error message with a line number, so in that sense it's good. The error message is a little bit different though, so for consistency's sake with the rest of Elixir it might be a good idea to keep it in.

Without this clause:
`key :age not found`
with this clause:
`key :age not found in: %Kernel.ErrorsTest.GoodStruct{name: \"john\"}`

Reply:
@derekkraan Yeah, I noticed that. I am not convinced that printing the struct is good though. It works fine for small structs but in practice we have large ones and they end-up adding too much noise.

Reply:
Ok. It's also not really necessary to find the error and fix it. I'll remove the clause and squash commits.